x-pack/plugins/security/server/routes/authorization/roles/get.ts

```diff
@@ -27,6 +28,9 @@ export function defineGetRolesRoutes({
       },
       validate: {
         params: schema.object({ name: schema.string({ minLength: 1 }) }),
+        query: schema.maybe(
+          schema.object({ replaceDeprecatedPrivileges: schema.maybe(schema.boolean()) })
+        ),
```

Comment:
__note:__ we need to make it optional so that, unlike the UI we control, clients using our APIs directly can retrieve exactly what they used to save the role, ensuring backward compatibility with any tooling or automation they have (what I save should be equal to what I retrieve). This could also be useful for debugging SDHs, allowing us to see both the original and mapped role definitions.